src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -18,19 +18,25 @@ namespace Microsoft.CST.OpenSource.PackageManagers
     using Version = SemanticVersioning.Version;
     using PackageUrl;
 
-    public class BaseProjectManager
+    public abstract class BaseProjectManager
     {
+        /// <summary>
+        /// The type of the project manager from the package-url type specifications.
+        /// </summary>
+        /// <seealso href="https://www.github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst"/>
+        public static readonly string Type = null!;
```

Comment:
Just wanted to note here, this **isn't** in use in this PR. It will be used in #318, and I anticipate it being useful in other places in the future as well.